Makefile

```diff
@@ -61,22 +61,22 @@ ifneq ($(HAS_SERVER),)
 endif
 
 ## Ensures NPM dependencies are installed without having to run this all the time.
-webapp/.npminstall:
+.PHONY: npminstall
+npminstall:
```

Comment:
Could we just change the target to `webapp/package.json` instead of writing custom functionality? Granted that would check the timestamp and not the hash, but that's probably good enough right?

Reply:
This sounds interesting -- can you elaborate on how it would be implemented? Today, we use the existence of the `webapp/.npminstall` file to skip the `npm install` check, but it's a one way-switch: there's no timestamp awareness. 

Is there a feature in `make` to tie the build event to a cached timestamp?

Reply:
Sorry. I think I misunderstand the point of this PR. It was already setup to only run `npm install` when `package.json` is changed. What is this PR trying to accomplish?

Reply:
This is the alternative: https://github.com/mattermost/mattermost-plugin-starter-template/pull/94
Does it work for you?

Reply:
Hah, yeah, that works! I suppose it's been too long since I've actually had to use make to intelligently manage dependency compilation :)

Also, that's an epic amount of compression relative to this pull request!